### ISSUE TITLE
Precompile fails on latest 0.7 - type Symbol has no field head

### ISSUE THREAD

**Issue by colbec** [JuliaStats/DataArrays.jl#257]
Something happened in the last 24 hours. DataArrays was fine yesterday for precompile but now fails with `type Symbol has no field head` on line 580 of operators.jl which shows `@eval begin`.

```
> versioninfo()
Julia Version 0.7.0-DEV.357
Commit 9e3318c984 (2017-05-25 19:37 UTC)
Platform Info:
  OS: Linux (x86_64-suse-linux)
  CPU: Intel(R) Core(TM) i5-4460  CPU @ 3.20GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, haswell)

> using CUDAdrv, CUDAnative, DataArrays, DataFrames, PyCall, PyPlot,  Images, ImageView
INFO: Precompiling module DataArrays.
ERROR: LoadError: LoadError: type Symbol has no field head
Stacktrace:
 [1] macro expansion at /home/colin/.julia/v0.7/DataArrays/src/operators.jl:580 [inlined]
 [2] anonymous at ./<missing>:?
 [3] include_from_node1(::String) at ./loading.jl:551
 [4] include(::String) at ./sysimg.jl:14
 [5] include_from_node1(::String) at ./loading.jl:551
 [6] include(::String) at ./sysimg.jl:14
 [7] anonymous at ./<missing>:2
while loading /home/colin/.julia/v0.7/DataArrays/src/operators.jl, in expression starting on line 579
while loading /home/colin/.julia/v0.7/DataArrays/src/DataArrays.jl, in expression starting on line 66
ERROR: Failed to precompile DataArrays to /home/colin/.julia/lib/v0.7/DataArrays.ji.
Stacktrace:
 [1] compilecache(::String) at ./loading.jl:685
 [2] _require(::Symbol) at ./loading.jl:490
 [3] require(::Symbol) at ./loading.jl:398

```

**Comment by ararslan** [comment]
I can reproduce this. I'm not exactly sure what the problem is and what changed in Base to make this happen, but it looks like it's `@dataarray_binary_scalar` that's triggering it within the loop beginning on line 579 of src/operators.jl.

**Comment by getzdan** [comment]
Reproduces for me as well, and after a little hunt, fixed by #259 .

**Comment by sbromberger** [comment]
I can reproduce as well.
